front/app/components/InitiativeForm/index.tsx

```diff
@@ -112,7 +118,15 @@ const InitiativeForm = ({
     topic_ids: array()
       .required(formatMessage(messages.topicEmptyError))
       .min(1, formatMessage(messages.topicEmptyError)),
-    cosponsor_ids: array().optional(),
+    ...(cosponsorsRequired && {
+      cosponsor_ids: array()
+        .required(formatMessage(messages.cosponsorsEmptyError))
+        .min(1, formatMessage(messages.cosponsorsEmptyError))
```

Comment:
Should it be `appConfiguration?.data.attributes.settings.initiatives?.cosponsors_number` and not 1?
```suggestion
        .min(appConfiguration?.data.attributes.settings.initiatives?.cosponsors_number, formatMessage(messages.cosponsorsEmptyError))
```